GL-4154: matoma.js updated for web ide

akashkska opened the pull request:
This pull request refines the logic for hiding "Web IDE" elements in the UI by making the selection more precise and robust. Instead of relying solely on XPath queries for spans, it now uses a combination of CSS selectors and targeted XPath expressions to ensure only relevant "Web IDE" buttons, links, and menu items are hidden, while avoiding accidental hiding of elements related to "Code". Logging has also been added for better debugging and traceability.

Improvements to element selection and hiding:

* Replaced the broad XPath-based approach with a set of specific CSS selectors (`webIdeSelectors`) to target "Web IDE" related buttons, links, and menu items for hiding, making the logic more maintainable and less error-prone.
* Added logic to ensure only standalone "Web IDE" elements (not merged with "Code") are hidden, preventing accidental hiding of combined UI elements.

Debugging and traceability:

* Introduced console logging to track which "Web IDE" elements are being hidden, aiding future debugging and verification efforts.

jamilshaikh07 committed:
matoma.js updated for web ide

bansodejoyce commented on gitmod/matomo.js:
```diff
@@ -32,19 +32,53 @@ var checkInterval = setInterval(() => Promise.resolve(fetch('/api/v4/user'))
 function hideThings () {
-  // Fetch the document that contains 'Web IDE' text
-  var webIde = document.evaluate("//span[contains(., 'Web IDE') or contains(., 'Open in Web IDE')]", document, null, XPathResult.ORDERED_NODE_SNAPSHOT_TYPE, null );
+  console.log('hideThings: Starting to hide Web IDE elements');
```
Can we remove the console?

bansodejoyce replied:
We add console for debugging and verification, once everything looks good we remove the console since it is ui facing

bansodejoyce reviewed (changes requested):
Remove the console, rest looks good

jamilshaikh07 committed:
GL-4154: Improve Web IDE hiding logic and remove debug logs

- Enhanced targeting to hide Web IDE options in Edit dropdowns
- Preserved Code functionality and legitimate edit options
- Removed console.log statements for production
- Added multiple detection strategies for better coverage

jamilshaikh07 committed:
GL-4154: Add MutationObserver and improved Edit dropdown targeting

- Enhanced Strategy 4 with better GitLab DOM selectors (data-testid)
- Added Strategy 5 with MutationObserver for dynamic dropdown content
- Fixed selector issues and improved Web IDE detection in Edit menus
- Target multiple possible dropdown menu patterns

jamilshaikh07 committed:
GL-4154: Fix overly aggressive hiding - preserve legitimate IDE options

- Make Web IDE detection much more specific (exact text matches only)
- Preserve Visual Studio Code and IntelliJ IDEA options in Code dropdown
- Only hide exact 'Web IDE' and 'Open in Web IDE' text matches
- Add safeguards to prevent hiding legitimate development tools
- Improve XPath selectors to be more precise

jamilshaikh07 committed:
GL-4154: Add debug logging to identify Web IDE hiding issues

- Added extensive console logging to track hideThings execution
- Log element counts and found Web IDE elements
- Added MutationObserver logging to see if dynamic content is detected
- Will help identify why Web IDE elements are not being hidden

jamilshaikh07 committed:
GL-4154: Remove debug logging - production ready version

- Removed all console.log statements for production deployment
- Confirmed Web IDE hiding working correctly in QA
- Preserves legitimate IDE options (VS Code, IntelliJ IDEA)
- Hides only Web IDE and Open in Web IDE options

jamilshaikh07 commented:
@bansodejoyce console.log removed also qa tested it, https://acquia.atlassian.net/browse/GL-4154
can you merge?
CC: @akashkska
